src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurStandardAccountingExtractCollectorBatchBuilder.java

```diff
@@ -0,0 +1,310 @@
+        }
+        
+        Integer fiscalYear = universityDateService.getFiscalYear(saeFileContents.getBatchDate());
+        if (fiscalYear == null) {
+            throw new RuntimeException("No fiscal year found for batch date: " + saeFileContents.getBatchDate().toString());
+        }
+        
+        collectorBatch.setRecordType(CuGeneralLedgerConstants.COLLECTOR_HEADER_RECORD_TYPE);
+        collectorBatch.setUniversityFiscalYear(fiscalYear.toString());
+        collectorBatch.setChartOfAccountsCode(chartCode);
+        collectorBatch.setOrganizationCode(highestLevelOrgCode);
+        collectorBatch.setTransmissionDate(saeFileContents.getBatchDate());
+        collectorBatch.setBatchSequenceNumber(Integer.valueOf(batchSequenceNumber));
+        collectorBatch.setCampusCode(campusCode);
+        collectorBatch.setMailingAddress(campusAddress);
+        collectorBatch.setEmailAddress(notificationEmail);
+        collectorBatch.setPersonUserID(notificationPerson);
+        collectorBatch.setPhoneNumber(notificationPhone);
+    }
+
+    protected void buildOriginEntries(List<ConcurStandardAccountingExtractDetailLine> saeLines) {
+        for (ConcurStandardAccountingExtractDetailLine saeLine : saeLines) {
+            String itemKey = buildItemKey(saeLine);
+            if (shouldProcessLine(itemKey, saeLine)
+                    && concurSAEValidationService.validateConcurStandardAccountingExtractDetailLine(saeLine)) {
+                OriginEntryFull originEntry = originEntries.computeIfAbsent(
+                        itemKey, (newKey) -> buildOriginEntryForExtractedLine(newKey, saeLine));
+                compareLineNonKeyFieldsAgainstEntry(itemKey, saeLine, originEntry);
+                addLineAmountToOriginEntry(saeLine, originEntry);
+            }
+        }
+    }
+
+    protected void updateCollectorBatchWithOriginEntries() {
+        for (OriginEntryFull originEntry : originEntries.values()) {
+            collectorBatch.addOriginEntry(originEntry);
+        }
+        
+        Integer totalRecords = Integer.valueOf(collectorBatch.getOriginEntries().size());
+        KualiDecimal totalAmount = collectorBatch.getOriginEntries()
+                .stream()
+                .map(OriginEntryFull::getTransactionLedgerEntryAmount)
+                .reduce(KualiDecimal.ZERO, KualiDecimal::add);
+        
+        collectorBatch.setTotalRecords(totalRecords);
+        collectorBatch.setTotalAmount(totalAmount);
+    }
+
+    protected boolean shouldProcessLine(String itemKey, ConcurStandardAccountingExtractDetailLine saeLine) {
+        switch (saeLine.getPaymentCode()) {
+            case ConcurConstants.PAYMENT_CODE_CASH :
+            case ConcurConstants.PAYMENT_CODE_UNIVERSITY_BILLED_OR_PAID :
+                return true;
+            case ConcurConstants.PAYMENT_CODE_PRE_PAID_OR_OTHER :
+                LOG.warn("Found a transaction with a Pre-Paid/Other payment code; it will not be processed. Item key: " + itemKey);

```

Comment:
should these logs be debug or info? I am trying to understand why they are warnings. Are these types of entries not expected in the SAE file?